pkg/controller/install/deployment.go

```diff
@@ -179,3 +174,39 @@ func (i *StrategyDeploymentInstaller) checkForDeployments(deploymentSpecs []Stra
 	}
 	return nil
 }
+
```

Comment:
I think we want to call `FindAnyDeploymentsMatchingLabels` and `cleanupOrphanedDeployments` in `CheckInstalled` as well?

Reply:
`checkInstalled` is only used as status check to see if something is missing or unverifiable. I feel it is dubious to run certain cleanup func inside a func that is only intended to verify something. Also, we should only clean up orphaned deployments after new deployments have been installed.

Reply:
You don't think it would be reasonable to rely on the label checks for `CheckInstalled` as well? It seems like it would be preferable to finding based on names